src/nl/rubensten/texifyidea/settings/TexifySettings.kt

```diff
@@ -34,9 +34,50 @@ class TexifySettings : PersistentStateComponent<TexifySettings> {
     // Index of selected combobox item
     var automaticQuoteReplacement = QuoteReplacement.NONE
 
+    /**
+     * internal list which stores the commands data
+     */
+    private val labelCommandsInternal: HashMap<String, LabelingCommandInformation> =
+            hashMapOf("\\label" to LabelingCommandInformation("\\label", 1, true))
+
+    /**
+     * Verify that any key in the list has a leading backslash
+     */
+    val labelCommands: Map<String, LabelingCommandInformation>
+        get() {
+            return labelCommandsInternal.mapKeys { it.key.addLeadingSlash() }
+        }
+
+    /**
+     * Do not use this field anywhere! It is only needed to persist the data of the commands to the settings.
+     * It maps any command to its string representation and on first load back to the object
+     */
+    var labelCommandsAsString: Map<String, String>
+        get() {
+            return labelCommandsInternal.mapValues { it.value.toSerializableString() }
+        }
+        set(value) {
+            value.forEach { (key, info) -> labelCommandsInternal[key] = LabelingCommandInformation.fromString(info) }
+        }
+
     override fun getState() = this
 
     override fun loadState(state: TexifySettings) {
         XmlSerializerUtil.copyBean(state, this)
     }
+
+    fun addCommand(cmd: LabelingCommandInformation) {
+        labelCommandsInternal[cmd.commandName] = cmd
```

Comment:
Why is information from `LabelingCommandInformation` duplicated in the key? `commandName` now appears in the key as wel as the value. Maybe leave it out of `LabelingCommandInformation`, or is there a logical reason to keep it in there?

Reply:
The reason to have it in `LabelingCommandInformation` is, that I sometimes only have this object.
The map is used to get the searched element in a fast way. I thought about a list to store the single elements, but I decided for a map, because in a map a search for an element is much faster.
It could be possible to remove it, but in my opinion it is more logical to have all informations about an element in one class.
In the settings dialog I map the whole row to this class, without the field for the key I would have to return two values or something else.
I hope you understand what I wanted to say, why I decided to do it this way.